include/units.h

```diff
@@ -2151,52 +2301,117 @@ namespace units
+	 *				to which both `unit` arguments can be converted to without requiring a division operation or truncating
+	 *				any value of these conversions, although floating-point units may have round-off errors.
+	 */
+	template<class UnitConversionLhs, class Tx, class UnitConversionRhs, class Ty, template<typename> class NonLinearScale>
+	struct common_type<units::unit<UnitConversionLhs, Tx, NonLinearScale>, units::unit<UnitConversionRhs, Ty, NonLinearScale>> :
+		std::enable_if<units::traits::is_convertible_unit_conversion_v<UnitConversionLhs, UnitConversionRhs>,
+			units::unit<
+				units::unit_conversion<
+					units::detail::ratio_gcd<typename UnitConversionLhs::conversion_ratio, typename UnitConversionRhs::conversion_ratio>,
+					units::traits::dimension_of_t<UnitConversionLhs>,
+					units::detail::ratio_gcd<typename UnitConversionLhs::pi_exponent_ratio, typename UnitConversionRhs::pi_exponent_ratio>,
+					units::detail::ratio_gcd<typename UnitConversionLhs::translation_ratio, typename UnitConversionRhs::translation_ratio>>,
+				common_type_t<Tx, Ty>,
+				NonLinearScale>> {};
+}
+
+namespace units
+{
+
 	//----------------------------------------
 	//	UNIT_T COMPOUND ASSIGNMENT OPERATORS
 	//----------------------------------------
 
-	template<class UnitConversion, typename T, template<typename> class NonLinearScale, typename RhsType>
-	inline constexpr unit<UnitConversion, T, NonLinearScale>& operator+=(unit<UnitConversion, T, NonLinearScale>& lhs, const RhsType& rhs) noexcept
+	/** @cond */	// DOXYGEN IGNORE
+	namespace detail
 	{
-		static_assert(traits::is_convertible_unit_v<unit<UnitConversion, T, NonLinearScale>, RhsType> ||
-			(traits::is_dimensionless_unit<decltype(lhs)>::value && std::is_arithmetic_v<RhsType>), 
-			"parameters are not compatible units.");
+		/**
+		 * @brief		Helper to make the use of a template parameter a non-deduced context.
+		 */
+		template<class T>
+		struct type_identity
+		{
+			using type = T;
+		};
 
+		template<class T>
+		using type_identity_t = typename type_identity<T>::type;
+	}
+	/** @endcond */	// END DOXYGEN IGNORE
+
+	template<class UnitConversion, typename T, template<typename> class NonLinearScale>
+	constexpr unit<UnitConversion, T, NonLinearScale>& operator+=(unit<UnitConversion, T, NonLinearScale>& lhs, const detail::type_identity_t<unit<UnitConversion, T, NonLinearScale>>& rhs) noexcept
+	{
 		lhs = lhs + rhs;
 		return lhs;
 	}
 
-	template<class UnitConversion, typename T, template<typename> class NonLinearScale, typename RhsType>
-	inline constexpr unit<UnitConversion, T, NonLinearScale>& operator-=(unit<UnitConversion, T, NonLinearScale>& lhs, const RhsType& rhs) noexcept
+	template<class UnitConversion, typename T, template<typename> class NonLinearScale>
+	constexpr unit<UnitConversion, T, NonLinearScale>& operator-=(unit<UnitConversion, T, NonLinearScale>& lhs, const detail::type_identity_t<unit<UnitConversion, T, NonLinearScale>>& rhs) noexcept

```

Comment:
what's `type_identity_t`?

Reply:
It's the identity metafunction. It's defined [here](https://github.com/nholthaus/units/pull/126/commits/87194b5854f89438fabf34888066e2a977d81a4c#diff-cc75003c399d878c360873c2c8c783d6R2306) and allows disabling template argument deduction. It's been proposed for standarization. The [paper](https://wg21.link/P0887) explains it well.